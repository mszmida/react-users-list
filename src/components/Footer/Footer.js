import React from 'react';

import styles from './Footer.module.css';
import reactLogo from '../../assets/logo.svg';

function Footer() {
  return (
    <>
      <footer className={styles.content}>
        <span>Powered by</span>

        <a href="https://reactjs.org" target="_blank" rel="noopener noreferrer">
          <img className={styles.logo} src={reactLogo} alt="React logo" />
        </a>
      </footer>
    </>
  );
}

export default Footer;
