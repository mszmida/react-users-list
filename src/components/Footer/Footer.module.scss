.content {
  text-align: center;
  padding: 20px 0;
}

.logo {
  animation: logo-spin infinite 20s linear;
  height: 5vmin;
  vertical-align: middle;
}

@keyframes logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
