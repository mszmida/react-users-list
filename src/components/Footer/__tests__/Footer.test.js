import React from 'react';
import { render } from '@testing-library/react';

import Footer from '../Footer';

describe('<Footer /> component', () => {
  test('matches the snapshot', () => {
    const { asFragment } = render(<Footer />);

    expect(asFragment()).toMatchSnapshot();
  });
});
