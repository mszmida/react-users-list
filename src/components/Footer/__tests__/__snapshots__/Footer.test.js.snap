// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> component matches the snapshot 1`] = `
<DocumentFragment>
  <footer
    class="content"
  >
    <span>
      Powered by
    </span>
    <a
      href="https://reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="React logo"
        class="logo"
        src="logo.svg"
      />
    </a>
  </footer>
</DocumentFragment>
`;
